transport/tchannel/channel_inbound.go

```diff
@@ -102,3 +103,13 @@ func (i *ChannelInbound) Stop() error {
 func (i *ChannelInbound) IsRunning() bool {
 	return i.once.IsRunning()
 }
+
+// Introspect returns the state of the inbound for introspection purposes.
+func (i *ChannelInbound) Introspect() introspection.InboundStatus {
```

Comment:
later maybe. as we discussed, this is internal to yarpc, the "outside world" introspection API might take some io.Writer eventually.